File: 28-find-the-index-of-the-first-occurrence-in-a-string/java/Solution.java
class Solution {
    public static int strStr(String haystack, String needle) {
        int haystackLength = haystack.length();
        int needleLength = needle.length();

        if (haystackLength < needleLength) {
            return -1;
        }

        for (int i = 0; i <= haystackLength - needleLength; i++) {
            if (haystack.substring(i, i + needleLength).equals(needle)) {
                return i;
            }
        }

        return -1;
    }
}

File: 28-find-the-index-of-the-first-occurrence-in-a-string/python/Solution.py
class Solution(object):
    @staticmethod
    def strStr(haystack: str, needle: str) -> int:
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        haystack_length = len(haystack)
        needle_length = len(needle)

        if haystack_length < needle_length:
            return -1

        for i in range(haystack_length - needle_length):
            if haystack[i:i + needle_length] == needle:
                return i

        return -1
